streaming/consumer.cc

```diff
@@ -65,11 +65,18 @@ reader_consumer_v2 make_streaming_consumer(sstring origin,
                         cf->enable_off_strategy_trigger();
                     }
                     return cf->add_sstable_and_update_cache(sst, offstrategy);
-                }).then([cf, s, sst, use_view_update_path, &vb]() mutable -> future<> {
+                }).then([cf, s, sst, use_view_update_path, &vb, origin] () mutable -> future<> {
                     if (!use_view_update_path) {
                         return make_ready_future<>();
                     }
-                    return vb.local().register_staging_sstable(sst, std::move(cf));
+                    return vb.local().register_staging_sstable(sst, std::move(cf)).then([sst = std::move(sst), origin = std::move(origin)] {
+                        if (origin != sstables::repair_origin) {
+                            return make_ready_future<>();
+                        }
+
+                        // for repair wait until the related view is updated
+                        return sst->wait_until_view_is_built();
```

Comment:
This will make repair wait on view update generation synchronously, after producing each sstable.
We want repair to continue and only wait up view update generation at the very end (of the current repair range).

Reply:
Ack. New version changes the approach.